angular2-jwt.ts

```diff
@@ -378,11 +371,21 @@ export class AuthModule {
     }
   }
 
-  static forRoot(config: AuthConfig): ModuleWithProviders {
+  static forRoot(config?: IAuthConfigOptional): ModuleWithProviders {
     return {
       ngModule: AuthModule,
       providers: [
-        {provide: AuthConfig, useValue: config}
+        {provide: AUTH_CONFIGURATION, useValue: config ? config : {}},
```

Comment:
Thinking about it more, you introduced it so you could do useFactory for AuthConfig. Is there a particular reason for that? It still makes more sense to me to make AuthConfig a useValue.